lib/config/fileconf.go

```diff
@@ -2561,12 +2561,14 @@ func (wds *WindowsDesktopService) Check() error {
 		return host.AD
 	})
 
-	if hasAD && wds.LDAP.Addr == "" {
+	ldapConfigOK := wds.LDAP.Addr != "" || (wds.LDAP.Domain != "" && wds.LDAP.LocateServer.Enabled)
```

Comment:
`Domain` is always required, whether you're using a hard-coded address or DNS location.